### PR TITLE
Expose cse execution time

### PR DESCRIPTION
**Pull request by adilliadil**

### CODE REVIEW

File: parts/linux/cloud-init/artifacts/cse_install.sh [hunk 1]
```diff
@@ -402,6 +398,18 @@ cleanUpContainerd() {
     rm -Rf $CONTAINERD_DOWNLOADS_DIR
 }
 
+configureGPUDrivers() {
+  if $FULL_INSTALL_REQUIRED; then
+        installGPUDrivers
+    fi
```
Comment (yizhang4321) [inline parts/linux/cloud-init/artifacts/cse_install.sh:404]: This "fi" is not aligned with its corresponding "if" above.

**Review by yizhang4321 — APPROVED** [review]

File: parts/linux/cloud-init/artifacts/cse_install.sh (hunk 1 above)
Comment (adilliadil) [reply parts/linux/cloud-init/artifacts/cse_install.sh]: Done

File: pkg/agent/testdata/AKSUbuntu1604+CustomKubeletConfig+CustomLinuxOSConfig/line16.sh [hunk 2]
```diff
@@ -2,10 +2,16 @@
 EXIT_CODE=$?
 systemctl --no-pager -l status kubelet >> /var/log/azure/cluster-provision-cse-output.log 2>&1
 OUTPUT=$(cat /var/log/azure/cluster-provision-cse-output.log | head -n 30)
+START_TIME=$(echo "$OUTPUT" | cut -d ',' -f -1 | head -1)
+CSE_EXECUTION_DURATION=$(echo $(($(date +%s) - $(date -d "$START_TIME" +%s))))
+echo "CSE Total Execution Duration:$CSE_EXECUTION_DURATION  seconds" >> /var/log/azure/cluster-provision-execution-durations.log 2>&1
```
Comment (xuto2) [inline pkg/agent/testdata/AKSUbuntu1604+CustomKubeletConfig+CustomLinuxOSConfig/line16.sh:7]: the total duration should be a separate property in the JSON_STRING. The content of /var/log/azure/cluster-provision-execution-durations.log cannot be directly put into json_string as the json_string has limited length. It's also not parsable by RP. Let's only expose the total time for now.

File: parts/linux/cloud-init/artifacts/cse_main.sh [hunk 3]
```diff
@@ -76,58 +76,48 @@ else
...
-    if $FULL_INSTALL_REQUIRED; then
-        installGPUDrivers
-    fi
-    ensureGPUDrivers
-    if [[ "${ENABLE_GPU_DEVICE_PLUGIN_IF_NEEDED}" = true ]]; then
-        systemctlEnableAndStart nvidia-device-plugin || exit $ERR_GPU_DEVICE_PLUGIN_START_FAIL
-    else
-        systemctlDisableAndStop nvidia-device-plugin
-    fi
+  run_and_log_execution_time configureGPUDrivers
```
Comment (xuto2) [inline parts/linux/cloud-init/artifacts/cse_main.sh:102]: would this work? configureGPUDrivers references ensureGPUDrivers

File: pkg/agent/testdata/AKSUbuntu1604+CustomKubeletConfig+CustomLinuxOSConfig/line16.sh (hunk 2 above)
Comment (xuto2) [reply pkg/agent/testdata/AKSUbuntu1604+CustomKubeletConfig+CustomLinuxOSConfig/line16.sh:7]: "CSEExecutionDurations": "cleanUpContainerd:0 seconds\ncleanUpGPUDrivers:0 seconds\ncleanUpContainerImages:0 seconds\ninstallContainerRuntime:2 seconds\ninstallNetworkPlugin:7 seconds\ninstallKubeletKubectlAndKubeProxy:0 seconds\ncreateKubeManifestDir:0 seconds\nconfigureK8s:0 seconds\nconfigureCNI:1 seconds\nensureDocker:23 seconds\nensureMonitorService:0 seconds\nensureSysctl:0 seconds\nensureKubelet:1 seconds\nensureJournal:1 seconds\nensureUpdateNodeLabels:0 seconds\nCSE Total Execution Duration:39 seconds"

Comment (xuto2) [reply pkg/agent/testdata/AKSUbuntu1604+CustomKubeletConfig+CustomLinuxOSConfig/line16.sh:7]: also cluster-provision-cse-output.log and the other new log file should be downloadable from asc
https://github.com/Azure/azure-diskinspect-service

**Pull request merged**